client/data/marketplace/use-site-search-es-query.ts

```diff
@@ -0,0 +1,300 @@
+
+const mapIndexResultsToPluginData = ( results: ESHits ): Plugin[] => {
+	return results.map( ( { fields: hit } ) => {
+		return {
+			name: hit.title_en, // TODO: add localization
+			slug: hit.slug,
+			version: hit.stable_tag,
+			author: createAuthorUrl(
+				hit[ 'meta.header_author.value' ],
+				hit[ 'meta.header_author_uri.value' ]
+			),
+			author_profile: hit[ 'meta.header_author_uri.value' ], // TODO: get author profile URL
+			tested: hit.tested,
+			rating: mapStarRatingToPercent( hit.rating ),
+			num_ratings: hit.num_ratings,
+			support_threads: hit.support_threads,
+			support_threads_resolved: hit.support_threads_resolved,
+			active_installs: hit.active_installs,
+			last_updated: hit[ 'meta.last_updated.value' ],
+			short_description: hit.excerpt_en, // TODO: add localization
+			icons: createIconsObject( hit.slug, hit[ 'meta.assets_icons.value' ] ),
+		};
+	} );
+};
+
+/**
+ *
+ * @param url The url to post to
+ * @param body The body of the request
+ * @returns {Promise<Response>} A promise that resolves to the response body
+ */
+async function postRequest( url: string, body: object ) {
+	const response = await fetch( `${ url }`, {
+		method: 'POST',
+		body: JSON.stringify( body ),
+		headers: { Accept: 'application/json', 'Content-Type': 'application/json' },
+	} );
+
+	if ( response.ok ) {
+		return await response.json();
+	}
+	throw new Error( await response.text() );
+}
+/**
+ * Based on WPORG plugin search scoring
+ * See https://github.com/WordPress/wordpress.org/blob/7f3e0d50e02008f5edd46d281052caf827661550/wordpress.org/public_html/wp-content/plugins/plugin-directory/libs/site-search/jetpack-search.php#L1040 for more info
+ */
+export const scoreQuery = ( query: object ) => {
+	const today = new Date();
+	const dateOrigin = today.toISOString().slice( 0, 10 );
+
+	return {
+		function_score: {
+			query,
+			functions: [
+				{
+					exp: {
+						plugin_modified: {
+							origin: dateOrigin,
+							offset: '180d',
+							scale: '360d',
+							decay: 0.5,
+						},
+					},
+				},
+				{
+					exp: {
+						tested: {
+							origin: '4.0',
+							offset: 0.1,
+							scale: 0.4,
+							decay: 0.6,
+						},
+					},
+				},
+				{
+					field_value_factor: {
+						field: 'active_installs',
+						factor: 0.375,
+						modifier: 'log2p',
+						missing: 1,
+					},
+				},
+				{
+					filter: {
+						range: {
+							active_installs: {
+								lte: 1000000,
+							},
+						},
+					},
+					exp: {
+						active_installs: {
+							origin: 1000000,
+							offset: 0,
+							scale: 900000,
+							decay: 0.75,
+						},
+					},
+				},
+				{
+					field_value_factor: {
+						field: 'support_threads_resolved',
+						factor: 0.25,
+						modifier: 'log2p',
+						missing: 0.5,
+					},
+				},
+				{
+					field_value_factor: {
+						field: 'rating',
+						factor: 0.25,
+						modifier: 'sqrt',
+						missing: 2.5,
+					},
+				},
+			],
+			score_mode: 'multiply',
+		},
+	};
```

Comment:
I've tried to replicate the same sorting criteria that it sent to the site search endpoint from the WPORG backend here: https://github.com/WordPress/wordpress.org/blob/7f3e0d50e02008f5edd46d281052caf827661550/wordpress.org/public_html/wp-content/plugins/plugin-directory/libs/site-search/jetpack-search.php#L1040

| Before | After | WPORG API |
| ------ | ----- | --- | 
| ![image](https://user-images.githubusercontent.com/11555574/165739180-0f208cfb-dbb8-425e-8429-4933e1537908.png) | ![image](https://user-images.githubusercontent.com/11555574/165738985-7a9d893c-33b7-40c7-81ea-de0bcfb41524.png) | ![image](https://user-images.githubusercontent.com/11555574/165739038-81f71ac6-266f-47af-90f9-278979857392.png) |